src/panels/config/entities/entity-registry-settings-editor.ts

```diff
@@ -1194,20 +1198,18 @@ export class EntityRegistrySettingsEditor extends LitElement {
     ) {
       // Change invert setting
       const origEntityId = this.entry.options?.switch_as_x?.entity_id;
-      // remove current helper
-      await deleteConfigEntry(this.hass, this.helperConfigEntry.entry_id);
-
       if (!origEntityId) {
         // should not happen, guard for types
       } else {
-        const configFlow = await createConfigFlow(this.hass, "switch_as_x");
-        const configFlowResult = (await handleConfigFlowStep(
+        const configFlow = await createOptionsFlow(
```

Comment:
Can we do the same for changing the switch as type? (Can be a different PR)

Reply:
I would prefer not, because when changing the switch as type we want to delete the entity registry item etc.